modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_lp_timer_zephyr.c

```diff
@@ -99,99 +72,76 @@ void nrf_802154_lp_timer_deinit(void)
 	nrf_802154_clock_lfclk_stop();
 }
 
-void nrf_802154_lp_timer_critical_section_enter(void)
+uint64_t nrf_802154_platform_sl_lptimer_current_lpticks_get(void)
 {
-	nrf_802154_sl_mcu_critical_state_t state;
-
-	nrf_802154_sl_mcu_critical_enter(state);
-
-	if (!m_in_critical_section) {
-		(void)z_nrf_rtc_timer_compare_int_lock(m_rtc_channel);
-		m_in_critical_section = true;
-	}
-
-	nrf_802154_sl_mcu_critical_exit(state);
+	return z_nrf_rtc_timer_read();
 }
 
-void nrf_802154_lp_timer_critical_section_exit(void)
+uint64_t nrf_802154_platform_sl_lptimer_us_to_lpticks_convert(uint64_t us, bool round_up)
 {
-	nrf_802154_sl_mcu_critical_state_t state;
-
-	nrf_802154_sl_mcu_critical_enter(state);
-
-	m_in_critical_section = false;
-
-	z_nrf_rtc_timer_compare_int_unlock(m_rtc_channel, m_is_running);
-
-	nrf_802154_sl_mcu_critical_exit(state);
+	if (round_up) {
+		return NRF_802154_SL_US_TO_RTC_TICKS(us);
+	} else {
+		return NRF_802154_SL_US_TO_RTC_TICKS(us) - 1ULL;
+	}
 }
 
-uint32_t nrf_802154_lp_timer_time_get(void)
+uint64_t nrf_802154_platform_sl_lptimer_lpticks_to_us_convert(uint64_t lpticks)
 {
-	/* Please note that this function does not handle RTC overflow */
-	return NRF_802154_SL_RTC_TICKS_TO_US(z_nrf_rtc_timer_read());
+	/* Calculations are performed on uint64_t as it is safe to assume
+	 * overflow will not occur in any foreseeable future.
+	 */
+	return NRF_802154_SL_RTC_TICKS_TO_US(lpticks);
 }
 
-uint32_t nrf_802154_lp_timer_granularity_get(void)
+void nrf_802154_platform_sl_lptimer_schedule_at(uint64_t fire_lpticks)
 {
-	return NRF_802154_SL_US_PER_TICK;
-}
+	/* This function is not required to be reentrant, hence no critical section. */
```

Comment:
But:
1. can be preempted by `rtc_irq_handler`
2. `rtc_irq_handler` can be preempted by `schedule_at`